From e670b4921b691921f1acd7207efbf888fe1dd023 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sun, 20 Mar 2022 17:18:32 +0200
Subject: [PATCH] [SW-76678] gaudi3: skip iatu config for slave die

There wont be need for any inbound access via slave die
as master die will be responsible for executing the recovery
mode. This patch skips iatu configuration for slave die.

Change-Id: I68a5b0ed3093f6e5a2124a266dc90afae076f94f
---

diff --git a/common/pcie.c b/common/pcie.c
index d028c5e..8a994a7 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -120,7 +120,8 @@
 		}
 
 		/* Configure IATU */
-		pcie_config_iatu();
+		if (!arcboot_rom_global.die_id)
+			pcie_config_iatu();
 
 		/* enable flr handling by HW state machine */
 		enable_flr_handling_fsm();
